src/types.ts

```diff
@@ -292,3 +292,29 @@ export interface UploadResult {
   /** Root ID in the proof set */
   rootId?: number
 }
+
+/**
+ * Proof set data returned from the API
+ */
+export interface ProofSetData {
+  /** The proof set ID */
+  id: number
+  /** Array of root data in the proof set */
+  roots: ProofSetRootData[]
+  /** Next challenge epoch */
+  nextChallengeEpoch: number
+}
+
+/**
+ * Individual proof set root data from API
+ */
+export interface ProofSetRootData {
+  /** Root ID within the proof set */
+  rootId: number
+  /** The root CID */
+  rootCid: string
+  /** Sub-root CID (usually same as rootCid) */
+  subrootCid: string
```

Comment:
```suggestion
  rootCid: string
  /** Sub-root CID (usually same as rootCid) */
  subrootCid: string
```

I wouldn't mind these converted to `CommP`

I think the best way to achieve this is to replace `validateProofSetData` _and_ `isProofSetData` with an `asProofSetData` and use the same pattern as the `asCommP` - that is, don't throw, just return `null` if it doesn't validate, but as you are validating, you are also building the new form, basically copying everything over except for these CIDs which you'll run `asCommP` on and get non-`null`.

So far (as far as I can remember), the validators don't do any conversion, so this is a new pattern, which is OK, just name it differently because you're not doing an `is` boolean, you're doing an `as` conversion which may or may not work.